plugins/webhooks/github.js

```diff
@@ -85,71 +96,72 @@ function pullRequestOpened(options, request, reply) {
-                job.archived = true;
+            return job.getRunningBuilds()
+                // Stop running builds
+                .then(builds => Promise.all(builds.map(stopRunningBuild)))
+                // update job
+                .then(() => {
+                    job.state = 'DISABLED';
+                    job.archived = true;
 
-                return job.update();
-            })
-    )
-    // log some stuff
-    .then(() => {
-        request.log(['webhook-github', eventId, jobId], `${name} disabled`);
+                    return job.update();
+                })
+                // log some stuff
+                .then(() => {
+                    request.log(['webhook-github', eventId, jobId], `${name} disabled`);
 
-        return reply().code(200);
-    })
-    // something went wrong
-    .catch(err => reply(boom.wrap(err)));
+                    return reply().code(200);
+                });
+        })
+        // something went wrong
+        .catch(err => reply(boom.wrap(err)));
 }
 
 /**
  * Stop any running builds and start the build for the synchronized pull-request
  * @method pullRequestSync
  * @param  {Object}       options
- * @param  {String}       options.eventId    Unique ID for this GitHub event
- * @param  {String}       options.jobId      Identifier for the Job
- * @param  {String}       options.name       Name of the job (PR-1)
- * @param  {String}       options.sha        specific SHA1 commit to start the build with
- * @param  {String}       options.username   User who created the PR
- * @param  {Hapi.request} request Request from user
- * @param  {Hapi.reply}   reply   Reply to user
+ * @param  {String}       options.eventId       Unique ID for this GitHub event
+ * @param  {String}       options.pipelineId    Identifier for the Pipeline
+ * @param  {String}       options.jobId         Identifier for the Job
+ * @param  {String}       options.name          Name of the job (PR-1)
+ * @param  {String}       options.sha           Specific SHA1 commit to start the build with
+ * @param  {String}       options.username      User who created the PR
+ * @param  {String}       options.prRef         Reference to pull request
+ * @param  {Pipeline}     options.pipeline      Pipeline model for the pr
+ * @param  {Hapi.request} request               Request from user
+ * @param  {Hapi.reply}   reply                 Reply to user
  */
 function pullRequestSync(options, request, reply) {
-    const buildFactory = request.server.app.buildFactory;
+    const jobFactory = request.server.app.jobFactory;
     const eventId = options.eventId;
     const name = options.name;
-    const username = options.username;
-    const sha = options.sha;
     const jobId = options.jobId;
 
-    return buildFactory.getBuildsForJobId({ jobId })
-        // stop all running builds
-        .then(builds => Promise.all(builds.map(stopRunningBuild)))
-        // log build stoppage
+    return jobFactory.get(jobId)
```

Comment:
Like `runMainJob` or `startPRJob` or whatever it gets named, the code for stopping all running builds in the PR is the same in both sync and closed handlers.